### ISSUE TITLE
All command interfaces are available in `inactive` state

### ISSUE THREAD

**Issue by mcbed** [ros-controls/ros2_control#931]
Hi, 
When switching the hardware interface into `inactive` state, all command interfaces remain available. I would expect that in this case, all states interfaces should remain `available` but all command interfaces should become `disabled` or at least  HW_IF_POSITION, HW_IF_VELOCITY, HW_IF_ACCELERATION, and HW_IF_EFFORT.  
Is there a reason to have it that way, or is it something that needs to be done yet ?

**Comment by destogl** [comment]
Yes, there are some things that have to be done, but they are quire simple, but not yet implemented. The concept is described in ros-controls/roadmap#51.

So, what has to be done:

- [ ] Decide what should be default behavior, i.e., command interfaces available only when `active` or also when `inactive`. I think we discussed that in our WG meetings, but I am not 100% sure what was the decision.
- [ ] Add support for a new flag into URDF parser, e.g., `only_available_when_active` or opposite if decided. Check the example here: https://github.com/ros-controls/roadmap/pull/51/files
- [ ] Don't change behavior at first but deprecate it, i.e., depending on the flag give warning that behavior will be changed. With this warning, we should also do backport to Humble – so that people change adjusting things timely.
- [ ] Decide on the flag if an interface should be added into available list of not in ResourceManager when HW component state is changed.

I hope I didn't miss any information here and the concept is clear what to do. I can support with reviews and tips anyone who wants to implement this.
